atc/exec/get_step_test.go

```diff
@@ -219,6 +220,12 @@ var _ = Describe("GetStep", func() {
 			Expect(resourceInstance.LockName("fake-worker")).To(Equal(expectedLockName))
 		})
 
+		It("secrets are tracked", func() {
```

Comment:
This test and its `get` + `task` equivalent feel a little too low-level and don't add much confidence that credentials are actually redacted.

In fact I was able to comment out the line that does the redacting and no tests under `atc/` failed. Can we make these tests a bit higher-level?

Reply:
I have added some test cases in `delete_factory_test.go`, so that redacting are tested. Now you should not be able to pass test if you comment out any line.